Code/Engine/Core/Utils/Blackboard.h

```diff
@@ -64,18 +64,28 @@ EZ_DECLARE_REFLECTABLE_TYPE(EZ_CORE_DLL, ezBlackboardEntryFlags);
 /// and then NPCs might use that information to make decisions.
 class EZ_CORE_DLL ezBlackboard : public ezRefCounted
 {
-private:
+public:
   ezBlackboard();
+  ~ezBlackboard();
 
-public:
-  /// \brief Factory method to create a new blackboard.
+  /// \brief Factory method to get access to a globally registered blackboard.
   ///
-  /// Since blackboards use shared ownership we need to make sure that blackboards are created in ezCore.dll.
-  /// Some compilers (MSVC) create local v-tables which can become stale if a blackboard was registered as global but the dll
-  /// which created the blackboard is already unloaded.
-  static ezSharedPtr<ezBlackboard> Create(ezAllocatorBase* pAllocator = ezFoundation::GetDefaultAllocator());
-  ~ezBlackboard();
+  /// If a blackboard with that name was already created globally before, its reference is returned.
```

Comment:
We still need a factory method for regular blackboards, otherwise you will run into the same problem as mentioned in the deleted comment if the plugin that created the blackboard is unloaded before the last reference.

Reply:
Okay

Reply:
I don't quite get the problem though. The allocator is in foundation, the constructor is in core, how can it generate code elsewhere?

Reply:
The keyword is "local vtable": https://groups.google.com/g/microsoft.public.vc.language/c/atSh_2VSc2w/m/EgJ3r_7OzVUJ?pli=1

Reply:
Just when you think you've seen it all...anyway, option 3 in his explanation would also be an option.

Reply:
Also, I guess this affects ALL shared objects. It's just more likely for blackboards.

Reply:
Yes it would also be an option but we don't do that anywhere else in the codebase and factory methods are pretty common in ez. Also it would unify the creation of global and regular blackboards. 
It affects all shared objects where the code lives in one dll but the creation happens in a plugin that can be unloaded. If the creation and code is in the same dll this problem doesn't occur. Although you can still run into problems if that dll is unloaded and another still has a reference, but atm I can't think of another problematic case. Haven't looked at the code though.

Reply:
Yes I actually agree that the creation method is the best solution. I'm wondering whether we should change this for all shared objects, just to be safe.

Reply:
Done. How did you actually discover this issue? There are several other shared classes (like all GAL objects) and it's hard to say whether any of them could also run into this problem.